### PR TITLE
feat: upgrade to ui@5 (TECH-387)

### PR DESCRIPTION
**Pull request by martinkrulltott**
Implements [TECH-387](https://jira.dhis2.org/browse/TECH-387)
Upgrades `ui-core` dep to `ui@5` (which has a lot of breaking changes)

Relates to https://github.com/dhis2/data-visualizer-app/pull/1068 and https://github.com/dhis2/dashboards-app/pull/891

**Major topics:**
- PeriodSelector refactored to use `name` again
- DynamicDimension refactored, note: won't work with pagination!
- New stories for Filter, PeriodDimension and DynamicDimension
- Any component using `Transfer`, `SingleSelect` or `MultiSelect` should've been updated

-----------------

**TODO**

- [x] Reimplement TransferOption icons

### CODE REVIEW

File: src/components/DataDimension/Detail.js [hunk 1]
```diff
@@ -18,14 +18,7 @@ export const Detail = ({ currentValue, onChange }) => {
...
                 label={i18n.t('Detail')}
-                selected={
-                    currentLabel
-                        ? {
-                              value: currentValue,
-                              label: currentLabel,
-                          }
-                        : null
-                }
+                selected={currentLabel ? currentValue : null}
```
Comment (jenniferarnesen) [inline src/components/DataDimension/Detail.js]: maybe it should be: ```{currentValue ? currentValue : null}```

File: src/components/DataDimension/Groups.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { SingleSelectField, SingleSelectOption } from '@dhis2/ui-core'
+import { SingleSelectField } from '@dhis2/ui'
```
Comment (jenniferarnesen) [inline src/components/DataDimension/Groups.js]: These can be combined into a single statement

File: src/components/DataDimension/Groups.js [hunk 3]
```diff
@@ -31,11 +32,7 @@ export const Groups = ({
             <div className="group-container">
                 <SingleSelectField
                     label={dataTypes[dataType].getGroupLabel()}
-                    selected={
-                        selected.id && selected.name
-                            ? { value: selected.id, label: selected.name }
-                            : {}
-                    }
+                    selected={selected.id && selected.name ? selected.id : null}
```
Comment (jenniferarnesen) [inline src/components/DataDimension/Groups.js]: Probably only need to check `selectedId` ?

Comment (martinkrulltott) [reply src/components/DataDimension/Groups.js]: Yes, actually `groupId` can be used directly

File: src/components/DataDimension/Detail.js (hunk 1 above)
Comment (martinkrulltott) [reply src/components/DataDimension/Detail.js]: My bad, `currentValue` can be used directly

File: src/components/DynamicDimension/ItemSelector.js [hunk 4]
```diff
@@ -49,31 +45,49 @@ const ItemSelector = ({
...
+                    label={label}
+                    highlighted={highlighted}
+                    onClick={onClick}
+                    onDoubleClick={onDoubleClick}
+                    value={value}
                     icon={GenericIcon}
                 />
-            ))}
-        </Transfer>
+            )}
```
Comment (jenniferarnesen) [inline src/components/DynamicDimension/ItemSelector.js]: You can spread the props:
```
renderOption={props => (
      <TransferOption {...props} icon={GenericIcon} />
)}
```

**Review by jenniferarnesen — APPROVED** [review]

**Pull request merged**